src/server.py

```diff
@@ -0,0 +1,32 @@
+"""Something server-y."""
+
+import socket
+
+
+def server():
+    """Start a server at localhost:9999."""
+    server = socket.socket( socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+    address = ('127.0.0.1', 5001)
+    server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    server.bind(address)
+
+    server.listen(1)
+
+    while True:
+
+        conn, addr = server.accept()
+
+        try:
+            message = ""
+            buffer_length = 10
+            while message[-3:] != 'EOF':
+                part = conn.recv(buffer_length)
+                message += part.decode('utf8')
```

Comment:
And anyway, for the purpose of this echo server, do you really need to decode the message at the server end, only to re-encode it and send it back?